crates/polars-core/src/chunked_array/ops/search.rs

```diff
@@ -0,0 +1,43 @@
+use polars_utils::float::IsFloat;
+
+use crate::prelude::*;
+
+impl<'a, T> ChunkSearch<'a, T::Native> for ChunkedArray<T>
+where
+    T: PolarsNumericType,
+{
+    fn index_of(&'a self, value: Option<T::Native>) -> Option<usize> {
+        // A NaN is never equal to anything, including itself. But we still want
+        // to be able to search for NaNs, so we handle them specially.
+        if value.map(|v| v.is_nan()) == Some(true) {
+            return self
+                .iter()
+                .position(|opt_val| opt_val.map(|v| v.is_nan()) == Some(true));
+        }
+
+        self.iter().position(|opt_val| opt_val == value)
+    }
+}
+
+impl ChunkSearch<'_, &Series> for ListChunked {
+    fn index_of(&self, value: Option<&Series>) -> Option<usize> {
```

Comment:
I really think we should stay away from `amortized` iter if we can. This is a very expensive way to implement this.

Reply:
Could you point me at an alternative approach? Normal `.iter()` seems like it'd be similarly expensive at first glance?

Reply:
I guess one could iterate over the leaf values array, and once a match is found figure out how many nulls preceded it. At least for a single level of nesting, with more that won't work and might have to use the current approach or something similar?

Reply:
I might decide to split list/array support out into separate PR, if this becomes a blocker, since it's still useful to support the other cases.

Reply:
Yes, let's start with the non-nested types, those will likely not be blockers indeed.